tests/5.1/teams/test_target_teams_thread_limit.c

```diff
@@ -0,0 +1,69 @@
+//===---------------- test_target_teams_thread_limit.c -----------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// This test uses the thread_limit clause on the teams construct. Specifically
+// testing if a thread_limit from a above target construct properly carries
+// down to the nested teams construct, as if it were directly on the construct
+// as defined in the spec. The test validates that only the specified 
+// threads are created by summing a shared variable across all threads 
+// (and teams). If the threads are correctedly limited this should produce the 
+// expected value. Additional warnings are sent if specific issues occur.
+//
+////===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int main() {
+	int errors = 0;
+	int shared = 0;
+	int num_teams = 0;
+
+	int testing_thread_limit = OMPVV_NUM_THREADS_DEVICE/OMPVV_NUM_TEAMS_DEVICE;
+
+	OMPVV_TEST_OFFLOADING;
+
+	#pragma omp target map(tofrom:num_teams,shared,errors) thread_limit(testing_thread_limit)
+	{
+		#pragma omp teams distribute parallel num_teams(OMPVV_NUM_TEAMS_DEVICE)
```

Comment:
`teams distribute parallel` does not seem to be a valid combined construct – as also GCC 12 complains.

Permitted are (cf. OpenMP 5.1):
* teams distribute parallel for/do
* teams distribute parallel for/do simd
And also:
* teams distribute
* teams distribute simd
* teams loop

In words of OpenMP 5.2:
* If _directive-name-A_ is **teams** then _directive-name-B_ may be **loop**, **distribute** or the directive name of a combined or composite construct for which _directive-name-A_ is distribute.
* If _directive-name-A_ is **distribute** then _directive-name-B_ may be **simd** or the directive name of a combined or composite construct for which _directive-name-A_ is **parallel** and **for** or **do** is a leaf construct.

The following works - but I am not sure that it does what you intend:
```suggestion
		#pragma omp teams num_teams(OMPVV_NUM_TEAMS_DEVICE)
		#pragma omp parallel
```

Reply:
Ultimately we are trying to test the thread limit construct so having it be two pragmas is totally ok for this test. Regardless will get this patched, thanks for the input!